src/content/reference/react/cache.md

```diff
@@ -0,0 +1,485 @@
+
+Notice that the first <CodeStep step={2}>`getData`</CodeStep> call does not `await` whereas the <CodeStep step={3}>second</CodeStep> does. [`await`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await) is a JavaScript operator that will wait and return the settled result of the promise. The first <CodeStep step={2}>`getData`</CodeStep> call simply initiates the `fetch` to cache the promise for the second <CodeStep step={3}>`getData`</CodeStep> to look-up.
+
+If by the <CodeStep step={3}>second call</CodeStep> the promise is still _pending_, then `await` will pause for the result. The optimization is that while we wait on the `fetch`, React can continue with computational work, thus reducing the wait time for the <CodeStep step={3}>second call</CodeStep>. 
+
+If the promise is already settled, either to an error or the _fulfilled_ result, `await` will return that value immediately. In both outcomes, there is a performance benefit.
+</DeepDive>
+
+<Pitfall>
+
+Calling a memoized function outside of a component will not use the cache.
+
+```jsx [[1, 3, "getUser"]]
+import {cache} from 'react';
+
+const getUser = cache(async (userId) => {
+  return await db.user.query(userId);
+});
+
+// 🚩 Wrong: Calling memoized function outside of render will not memoize.
+getUser('demo-id');
+
+async function DemoProfile() {
+  // ✅ Good: `getUser` will memoize.
+  const user = await getUser('demo-id');
+  return <Profile user={user} />;
+}
+```
+
+React only provides cache access to the memoized function in a component. When calling <CodeStep step={1}>`getUser`</CodeStep> outside of a component, it will still evaluate the function but not read or update the cache.
+
+This is because cache access is provided through a [context](/learn/passing-data-deeply-with-context) which is only accessibile from a component. 
+
+</Pitfall>
+
+<DeepDive>
+
+#### When should I use `cache`, [`memo`](/reference/react/memo), or [`useMemo`](/reference/react/useMemo)? {/*cache-memo-usememo*/}
+
+All mentioned APIs offer memoization but the difference is what they're intended to memoize, who can access the cache, and when their cache is invalidated.
+
+#### `useMemo` {/*deep-dive-use-memo*/}
+
+In general, you should use [`useMemo`](/reference/react/useMemo) for caching a expensive computation in a Client Component across renders. As an example, to memoize a transformation of data within a component.
+
+```jsx {4}
+'use client';
+
+function WeatherReport({record}) {
+  const avgTemp = useMemo(() => calculateAvg(record)), record);
+  // ...
+}
+
+function App() {
+  const record = getRecord();
+  return (
+    <>
+      <WeatherReport record={record} />
+      <WeatherReport record={record} />
+    </>
+  );
+}
+```
+In this example, `App` renders two `WeatherReport`s with the same record. Even though both components do the same work, they cannot share work. `useMemo` is only a local cache to the component instance. 
```

Comment:
> `useMemo` is only a local cache to the component instance. 

nit: this wording is a little awkward, maybe try something like "`useMemo`'s cache is local to the component"?